### ISSUE TITLE
Multiple workloads with the same image repo + webhooks

### ISSUE THREAD

**Issue by leosunmo** [keel-hq/keel#253]
I am facing a problem where my organisation uses the same docker image repo + tag for multiple different workloads (different execution modes for the same app/docker image). This means that if we use the [native webhook](https://keel.sh/v1/guide/documentation.html#Webhooks) feature we can only pass along an image repo and tag which would then presumably update all Keel managed deployment with that image repo. 

Is there any way to specify the specific workload I want Keel to deploy when I send a webhook? Ideally we would have an additional field in the webhook payload (maybe another "advanced" webhook alongside the simple one?) that would specify the workload type and name ("workload": "deployment/my-app").

Am I missing something obvious here? Is there a solution already in-place for this?

**Comment by rusenask** [comment]
As I understand, you have multiple deployments that use the same image and all have keel update policy, right? 

Yeah, maybe our native webhook could take additional selector as a hint.

I have a similar setup, where I start the same container with different commands, therefore I get those several workloads updated together. It's an updated image image after all, the only thing that slightly bothers me in this case is that most of the time only one part or the other has meaningful changes so there are, indeed, unnecessary restarts involved :) 

What registry are you using? How many different workloads are there with the same image?

**Comment by leosunmo** [comment]
Yes, we have multiple deployments with the same image that we'd like Keel to update when we send webhooks to it.

This is a pattern repeated throughout our infrastructure quite a lot so is definitely a must-have.

We're using ECR as our registry and therefore have our CI system call an AWS Lambda that then calls Keel's native webhook system (gives us auth to the cluster as well as makes sure we only expose Keel internally as the lambda is in the private network).

**Comment by rusenask** [comment]
Seems like it would be nice to have an HTTP API that just tells which deployment in which namespace to update then as your Lambda has already figured it out, what needs to change :) Are you using k8s or helm provider in keel?

**Comment by leosunmo** [comment]
An HTTP API sounds like it would be a great feature that would help out a lot.

I'm using the k8s provider, no Helm at the moment.
